backend/matching-service/matching_service/api.py

```diff
@@ -1,25 +1,68 @@
-from fastapi import FastAPI
+from fastapi import FastAPI, HTTPException
 from fastapi.middleware.cors import CORSMiddleware
 from structlog import get_logger
+from pydantic import BaseModel
+from redis import Redis
+from typing import List
 
 from .config import settings
+from matching_service.common import MatchRequest
 
 logger = get_logger()
 
+# Initialize FastAPI app
 app = FastAPI(
     title="Matching Service Backend",
 )
 
+# CORS middleware
 app.add_middleware(
     CORSMiddleware,
     allow_origins=settings.ORIGINS,
     allow_credentials=True,
     allow_methods=["*"],
-    # To update if after auth
     allow_headers=["*"],
 )
 
+# Define the Redis client
+redis_client = Redis.from_url("redis://localhost:6379/0")
 
-@app.get("/")
-async def index():
+def request_match(publisher: Redis, user: MatchRequest):
+    channel = RedisSettings.Channels.REQUESTS.value
+    publisher.publish(channel, user.model_dump_json())
+    logger.info(f"CLIENT: User {user.user} requested match for {user.topic}, {user.difficulty}")
+
+
+
+# Define the request model
+class User(BaseModel):
+    user: str
+    topic: str
+    difficulty: str
+
+
+@app.get("/match", response_model=dict)
+async def get_match(user: User):
+    """
+    Example GET endpoint for basic testing.
+    """
     return {"message": "Hello from matching service"}
+
+
+@app.post("/match", response_model=dict)
+async def create_match(user: User):
```

Comment:
The `User` model is redundant cos its the same as `MatchRequest` so it can just be `create_match(req: MatchRequest)`

Then, in ur try block u dont have to make a MatchRequest object